src/roiextractors/extractors/tiffimagingextractors/brukertiffimagingextractor.py

```diff
@@ -0,0 +1,123 @@
+
+        with tifffile.TiffFile(self.folder_path / self._file_paths[0], _multifile=False) as tif:
+            self._height, self._width = tif.pages.first.shape
+            self._dtype = tif.pages.first.dtype
+
+    def _get_xml_root(self):
+        xml_file_path = self.folder_path / f"{self.folder_path.stem}.xml"
+        assert Path(xml_file_path).is_file(), f"The XML configuration file is not found at '{self.folder_path}'."
+        tree = ElementTree.parse(xml_file_path)
+        return tree.getroot()
+
+    def _get_xml_metadata(self):
+        root = self._get_xml_root()
+        xml_metadata = dict()
+        xml_metadata.update(**root.attrib)
+        for child in root.findall(".//PVStateValue"):
+            metadata_root_key = child.attrib["key"]
+            if "value" in child.attrib:
+                if metadata_root_key in xml_metadata:
+                    continue
+                xml_metadata[metadata_root_key] = child.attrib["value"]
+            else:
+                xml_metadata[metadata_root_key] = []
+                for indexed_value in child:
+                    if "value" not in indexed_value.attrib:
+                        for subindexed_value in indexed_value:
+                            if "description" in subindexed_value.attrib:
+                                xml_metadata[metadata_root_key].append(
+                                    {subindexed_value.attrib["description"]: subindexed_value.attrib["value"]}
+                                )
+                            else:
+                                xml_metadata[child.attrib["key"]].append(
+                                    {indexed_value.attrib["index"]: subindexed_value.attrib["value"]}
+                                )
+                    if "description" in indexed_value.attrib:
+                        xml_metadata[child.attrib["key"]].append(
+                            {indexed_value.attrib["description"]: indexed_value.attrib["value"]}
+                        )
+                    elif "value" in indexed_value.attrib:
+                        xml_metadata[child.attrib["key"]].append(
+                            {indexed_value.attrib["index"]: indexed_value.attrib["value"]}
+                        )
+        return xml_metadata
+
+    def _get_files_names(self):
+        return [file.attrib["filename"] for file in self._get_xml_root().findall(".//File")]
+
+    def get_image_size(self) -> Tuple[int, int]:
+        return self._height, self._width
+
+    def get_num_frames(self) -> int:
+        return len(self._file_paths)
+
+    def get_sampling_frequency(self) -> float:
+        frame_period = float(self.xml_metadata["framePeriod"])
+        return 1 / frame_period
+
+    def get_channel_names(self) -> list:
+        channel_names = [file.attrib["channelName"] for file in self._get_xml_root().findall(".//File")]
+        unique_channel_names = list(set(channel_names))
+        return unique_channel_names
```

Comment:
Is this actually multi-channel data?

Reply:
The data that we currently has only one channel and is called `"Ch2"`. Since I'm not handling multi-channel data in `get_video()` I guess I can replace `.findall` to `.find`, and then it returns the first entry.

Reply:
`.findall` is fine to leave as a generality